lib/pact/consumer_contract/interaction_v2_parser.rb

```diff
@@ -4,13 +4,19 @@
 require 'pact/symbolize_keys'
 require 'pact/matching_rules'
 require 'pact/errors'
+require 'rack/utils'
 
 module Pact
   class InteractionV2Parser
 
     include SymbolizeKeys
 
     def self.call hash, options
+      if hash['request'].has_key? 'query'
+      params = Rack::Utils.parse_nested_query(hash['request']['query'])
+      query = URI.encode_www_form(params)
+      hash['request']['path'] = hash['request']['path'] << '?' << query
```

Comment:
Don't need this line, or line 17.

Reply:
You need `hash['request']['query'] = params`

Reply:
Sorry, I thought it needed to be converted into a string and added to the path! When it tested it against pact-mock_service, it was like pathception (query path added onto the path by my change, plus the existing query read and added to the path) - hence the need for tests =D

I had read your initial thoughts from and didn't realise we needed it to be encoded from a string, into a hash!